test/framework/cluster_proxy.go

```diff
@@ -285,6 +285,7 @@ func (p *clusterProxy) GetWorkloadCluster(ctx context.Context, namespace, name s
 // CollectWorkloadClusterLogs collects machines and infrastructure logs and from the workload cluster.
 func (p *clusterProxy) CollectWorkloadClusterLogs(ctx context.Context, namespace, name, outputPath string) {
 	if p.logCollector == nil {
+		fmt.Printf("Failed to get logs for workload Cluster %s: log collector is nil", klog.KRef(namespace, name))
```

Comment:
```suggestion
		fmt.Printf("Unable to get logs for workload Cluster %s: log collector not set", klog.KRef(namespace, name))
```
If that is a regular non-error case. I wonder if some providers simply don't implement that and then Unable might be more apprioriate than failed

(but up to you)